DigitalLearningSolutions.Data/DataServices/UserDataService.cs

```diff
@@ -340,6 +346,44 @@ public void ApproveDelegateUsers(params int[] ids)
             );
         }
 
+        public void RemoveDelegateUser(int delegateId)
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                var existingSessions = connection.Query<int>(
+                    @"SELECT SessionID FROM Sessions WHERE CandidateID = @delegateId",
+                    new { delegateId }
+                );
+
+                if (existingSessions.Any())
+                {
+                    throw new UserAccountInvalidStateException(
+                        $"Delegate user id {delegateId} cannot be removed as they have already started a session."
+                        );
+                }

```

Comment:
Oh sorry - I thought we'd change this at a db level, but at the same time, I think it might be worth getting the current state merged and then sorting it out in a different MR as Alex's ticket uses a few bits from here. (changing the DELETE_RULE for the Sessions table in the db - can we do it via a migration?)